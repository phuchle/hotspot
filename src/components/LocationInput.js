import React, { Component } from 'react';
import { Form, FormControl, Button } from 'react-bootstrap';
import PropTypes from 'prop-types';

class LocationInput extends Component {
  constructor(props) {
    super(props);

    this.state = {
      destination: '',
      location: ''
    };

    this.handleDestinationChange = this.handleDestinationChange.bind(this);
    this.handleLocationChange = this.handleLocationChange.bind(this);
  }
  handleDestinationChange(event) {
    this.setState({
      destination: event.target.value
    });
  }
  handleLocationChange(event) {
    this.setState({
      location: event.target.value
    });
  }
  render() {
    return (
      <Form
        inline
        style={this.props.formStyle}
        onSubmit={(event) => {
          event.preventDefault();
          this.props.handleSubmit(this.state);
        }}
      >
        <FormControl
          type="text"
          placeholder="Drinks"
          value={this.state.destination}
          onChange={this.handleDestinationChange}
          style={{marginRight: '10px'}}
        />
        <FormControl
          type="text"
          placeholder="San Jose, CA"
          value={this.state.location}
          onChange={this.handleLocationChange}
          style={{marginRight: '10px'}}
        />
        <Button
          type="submit"
          bsStyle="success"
        >
          Search!
        </Button>
      </Form>
    );
  }
}

LocationInput.propTypes = {
  formStyle: PropTypes.object,
  handleSubmit: PropTypes.func.isRequired,
};

export default LocationInput;
