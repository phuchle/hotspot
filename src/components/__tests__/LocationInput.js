import React from 'react';
import renderer from 'react-test-renderer';
import LocationInput from '../LocationInput';

describe ('LocationInput', () => {
  test('renders without crashing', () => {
    const component = renderer.create(
      <LocationInput handleSubmit={jest.fn()} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
