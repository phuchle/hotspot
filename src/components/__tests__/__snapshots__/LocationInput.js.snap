// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationInput renders without crashing 1`] = `
<form
  onSubmit={[Function]}
  style={undefined}
>
  <input
    className="form-control"
    id={undefined}
    onChange={[Function]}
    placeholder="Drinks"
    style={
      Object {
        "marginRight": "10px",
      }
    }
    type="text"
    value=""
  />
  <input
    className="form-control"
    id={undefined}
    onChange={[Function]}
    placeholder="San Jose, CA"
    style={
      Object {
        "marginRight": "10px",
      }
    }
    type="text"
    value=""
  />
  <button
    className="btn btn-success"
    disabled={false}
    type="submit"
  >
    Search!
  </button>
</form>
`;
